src/main/java/uk/ac/ebi/embl/converter/fftogff3/FFGFF3SourceAttributesFactory.java

```diff
@@ -0,0 +1,31 @@
+package uk.ac.ebi.embl.converter.fftogff3;
+
+import uk.ac.ebi.embl.api.entry.Entry;
+import uk.ac.ebi.embl.api.entry.feature.Feature;
+import uk.ac.ebi.embl.api.entry.qualifier.Qualifier;
+import uk.ac.ebi.embl.converter.IConversionRule;
+import uk.ac.ebi.embl.converter.gff3.GFF3SourceMetadata;
+
+public class FFGFF3SourceAttributesFactory implements IConversionRule<Entry, GFF3SourceMetadata> {
+
+    @Override
+    public GFF3SourceMetadata from(Entry entry) throws ConversionError {
+
+        for (Feature feature : entry.getFeatures().stream().sorted().toList()) {
```

Comment:
Suggestion to  get the source Feature using 
`Feature sourceFeature = entry.getPrimarySourceFeature();`